src/libutil/thread-pool.cc

```diff
@@ -150,7 +150,10 @@ void ThreadPool::doWork(bool mainThread)
         }
 
         didWork = true;
-    }
+        return Continue;
+    });
+    // We've been interrupted or there's no more work to do; either way we
+    // must return now.
```

Comment:
Doesn't this create the possibility that if there is an interrupt, the `ThreadPool` will return successfully? I.e. `pool.process()` will return without an error, even though not all of the work items have been executed. And then (for instance) the entire Nix command could succeed if the calling thread doesn't check for interrupts.